backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go

```diff
@@ -522,38 +527,67 @@ func (l *learnAndDetectBFLA) commandsRunner(ctx context.Context, command Command
 	return nil
 }
 
-func (l *learnAndDetectBFLA) validateAuthzModel(ctx context.Context, m AuthorizationModel, apiID uint) (AuthorizationModel, error) {
+func logDebugAuthModel(m AuthorizationModel) {
+	if !log.IsLevelEnabled(log.DebugLevel) {
+		return
+	}
+
 	jmodel, err := json.Marshal(m)
 	if err != nil {
 		log.Errorf("unable to marshal auth model %v", err)
 	}
-	log.Debugf("updated auth model:%s\n", jmodel)
+	log.Debugf("%s", jmodel)
+}
+
+func (l *learnAndDetectBFLA) mergeAuthzModel(ctx context.Context, newModel AuthorizationModel, oldModel AuthorizationModel, apiID uint) (AuthorizationModel, error) {
 	apiInfo, err := l.bflaBackendAccessor.GetAPIInfo(ctx, apiID)
 	if err != nil {
-		return m, fmt.Errorf("unable to get api info: %w", err)
+		return oldModel, fmt.Errorf("unable to get api info: %w", err)
 	}
 	tags, err := ParseSpecInfo(apiInfo)
 	if err != nil {
-		return m, fmt.Errorf("unable to parse spec info: %w", err)
+		return oldModel, fmt.Errorf("unable to parse spec info: %w", err)
 	}
 	specType := SpecTypeFromAPIInfo(apiInfo)
 	if specType == SpecTypeNone {
-		return m, fmt.Errorf("spec not present cannot process authorization model; apiID=%d", apiID)
+		return oldModel, fmt.Errorf("spec not present cannot process authorization model; apiID=%d", apiID)
 	}
-	for _, op := range m.Operations {
+
+	audienceMap := extractAudienceMap(oldModel)
+
+	for _, op := range newModel.Operations {
 		if op.Path == "" || op.Method == "" {
-			return m, fmt.Errorf("invalid auth model operation: %v. apiID=%d", op, apiID)
+			return oldModel, fmt.Errorf("invalid auth model operation: %v. apiID=%d", op, apiID)
 		}
 		op.Tags = resolveTagsForPathAndMethod(tags, op.Path, op.Method)
 
 		for audIdx, aud := range op.Audience {
 			if !aud.External {
 				if aud.K8sObject == nil ||
-					aud.K8sObject.Name == "" ||
-					aud.K8sObject.Uid == "" ||
-					aud.K8sObject.ApiVersion == "" ||
-					aud.K8sObject.Kind == "" {
-					return m, fmt.Errorf("invalid auth model audience for operation %v: [%d] %v . apiID = %d", op, audIdx, aud, apiID)
+					aud.K8sObject.Name == "" {
+					return oldModel, fmt.Errorf("invalid auth model audience for operation %v: [%d] %v . apiID = %d", op, audIdx, aud, apiID)
+				}
+				if aud.K8sObject.ApiVersion == "" || aud.K8sObject.Kind == "" || aud.K8sObject.Uid == "" {
+
+					nsMap, ok := audienceMap[aud.K8sObject.Name]
+					if !ok {
+						return oldModel, fmt.Errorf("unable to find the audience entry for %v, %v, name = %s, namespace = %s: name not found", op.Path, op.Method, aud.K8sObject.Name, aud.K8sObject.Namespace)
+					}
+					foundAud, ok := nsMap[aud.K8sObject.Namespace]
+					if !ok {
+						if aud.K8sObject.Namespace == "" && len(nsMap) == 1 {
+							for _, foundAud = range nsMap {
+								break
+							}
```

Comment:
I don't get it

Reply:
Get the first (and only) element in the map .